### PR TITLE
Remove dead peers more agressively

### LINKED ISSUE #3017: Peer connections time out
On `ae_uat` the node is not able to sync, the log is full of connection timeouts:

```
2019-11-18 10:10:08.364 [debug] <0.1560.0>@aec_peers:schedule_connect:730 Waiting 100 ms for the next connection
2019-11-18 10:10:08.465 [debug] <0.1560.0>@aec_peers:start_connection:760 Peer "pp_2qiq4Xa...oUVaCTd" - starting connection to 54.251.156.208:3015
2019-11-18 10:10:08.466 [debug] <0.1560.0>@aec_peer_connection:connect:72 New peer connection to #{host => <<"54.251.156.208">>,port => 3015,pubkey => <<213,124,221,120,255,112,249,23,125,189,240,246,149,239,62,65,86,231,125,43,54,197,117,158,54,108,226,236,233,242,13,86>>,r_pubkey => <<242,82,209,12,161,123,89,137,156,240,161,37,143,107,133,194,229,216,240,230,253,19,102,95,130,111,74,93,82,210,112,65>>}
2019-11-18 10:10:08.467 [debug] <0.1560.0>@aec_peers:schedule_connect:730 Waiting 2000 ms for the next connection
2019-11-18 10:10:09.467 [debug] <0.6818.5>@aec_peer_connection:handle_info:284 Failed to connected to {<<"54.251.156.208">>,3015}: {error,timeout}
2019-11-18 10:10:09.468 [debug] <0.1560.0>@aec_peers:on_connection_failed:821 Peer "pp_2qiq4Xa...oUVaCTd" - connection to 54.251.156.208:3015 failed by process <0.6818.5>
2019-11-18 10:10:09.468 [debug] <0.1560.0>@aec_peers:schedule_connect:730 Waiting 100 ms for the next connection
```
From the erlang shell:
```
(aeternity@localhost)9> aec_peers:connected_peers().
[]
(aeternity@localhost)10> length(aec_peers:available_peers()).
3030
```
It looks like there are too many available peers. The list includes also peers that are not running anymore and so are unreachable (so connections to them time out).

### PR DESCRIPTION
**Pull request by jur0**
Closes #3017

### CODE REVIEW

File: apps/aecore/src/aec_peers.erl [hunk 1]
```diff
@@ -74,12 +74,11 @@
 -define(DEFAULT_PING_INTERVAL,        120 * 1000). %  2 minutes
 -define(DEFAULT_UNBLOCK_INTERVAL, 15 * 60 * 1000). % 15 minutes
 -define(DEFAULT_MAX_INBOUND,                 100).
--define(DEFAULT_MAX_OUTBOUND,                 10).
+-define(DEFAULT_MAX_OUTBOUND,                 20).
```
Comment (dincho) [inline apps/aecore/src/aec_peers.erl]: please change the default value in the config schema as well

File: apps/aecore/src/aec_peers_pool.erl [hunk 2]
```diff
@@ -343,7 +342,7 @@ address_group({A, B, _, _}) -> <<A:8, B:8>>.
 %%    a peer should be put on standby when rejected; if rejected more than
 %%    the size of the given list due to `max_rejections' being larger than
 %%    the list, the last value of the list is used multiple times.
-%%    Default: `[5, 15, 30, 60, 120, 300, 600]'.</li>
+%%    Default: `[5, 15]'.</li>
```
Comment (tolbrino) [inline apps/aecore/src/aec_peers_pool.erl]:
```suggestion
%%    Default: `[5000, 15000]'.</li>
```

**Comment by jur0** [comment]
We need to deploy these changes to testnet, but if there are some old nodes, they still can gossip dead peers using the old params, our simulator doesn't simulate this scenario (or at least I'm not aware of it). I'm still in process of doing simulations...
The issue we had was with a node that lost connection to all peers and ended up with 1000s of peers (99% dead) trying to connect to them - it took couple of hours to find a peer which was alive.

**Comment by jur0** [comment]
Closing as this PR is split between config part (#3074) and another PR that will change the defaults.

**Pull request closed without merging**